internal/pkg/manager/nodestatus/nodestatus.go

```diff
@@ -0,0 +1,286 @@
+	pbv1alpha1 "sigs.k8s.io/security-profiles-operator/api/profilebase/v1alpha1"
+	seccompv1alpha1 "sigs.k8s.io/security-profiles-operator/api/seccompprofile/v1alpha1"
+	statusv1alpha1 "sigs.k8s.io/security-profiles-operator/api/secprofnodestatus/v1alpha1"
+	selinuxv1alpha1 "sigs.k8s.io/security-profiles-operator/api/selinuxpolicy/v1alpha1"
+	"sigs.k8s.io/security-profiles-operator/internal/pkg/config"
+	"sigs.k8s.io/security-profiles-operator/internal/pkg/util"
+)
+
+const (
+	reconcileTimeout = 1 * time.Minute
+	dsWait           = 30 * time.Second
+)
+
+var (
+	ErrNoOwnerProfile  = errors.New("No owner profile defined for this status")
+	ErrUnkownOwnerKind = errors.New("the node status owner is of an unknown kind")
+)
+
+// A StatusReconciler monitors node changes and updates the profile status.
+type StatusReconciler struct {
+	client client.Client
+	log    logr.Logger
+	record event.Recorder
+}
+
+func NewStatusReconciler(cli client.Client, log logr.Logger, record event.Recorder) *StatusReconciler {
+	return &StatusReconciler{
+		client: cli,
+		log:    log,
+		record: record,
+	}
+}
+
+// Security Profiles Operator RBAC permissions to manage SelinuxPolicy
+// nolint:lll
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=selinuxpolicies,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=selinuxpolicies/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=selinuxpolicies/finalizers,verbs=delete;get;update;patch
+
+// Security Profiles Operator RBAC permissions to manage SeccompProfile
+// nolint:lll
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=seccompprofiles,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=seccompprofiles/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=seccompprofiles/finalizers,verbs=delete;get;update;patch
+
+// Security Profiles Operator RBAC permissions to manage Node Statuses
+// nolint:lll
+// +kubebuilder:rbac:groups=security-profiles-operator.x-k8s.io,resources=securityprofilenodestatuses,verbs=get;list;watch
+
+// Reconcile reconciles a NodeStatus.
+func (r *StatusReconciler) Reconcile(_ context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.log.WithValues("nodeStatus", req.Name, "namespace", req.Namespace)
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	logger.Info("status change")
+
+	// get the status to be reconciled
+	instance := &statusv1alpha1.SecurityProfileNodeStatus{}
+	if err := r.client.Get(ctx, req.NamespacedName, instance); err != nil {
+		// Expected to find a node profile, return an error and requeue
+		// TODO(jhrozek): ignore not found?
+		return reconcile.Result{}, errors.Wrap(util.IgnoreNotFound(err), "Cannot find the node status")
```

Comment:
Not sure `IgnoreNotFound` is right here, since it will convert the error to nil and then errors.Wrap will be wrapping nil instead of an error. If it's returning an error either way then may as well keep the full error context.